### PR TITLE
Added COSV files upload button

### PR DESCRIPTION
**Pull request by Cheshiriks**
Снимок

Снимок2

Снимок3

Снимок4

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ val uploadVulnerabilityView: VFC = VFC {
                             className = ClassName("needs-validation")
 
                             fileManagerComponent {
-                                // todo: storage size shouldn't be more then 500MB per project
+                                // FixMe: need to add the correct directory for COSV files
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt:48]: not directory -- we have a dedicated API to upload COSV

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/OrganizationDto.kt [hunk 2]
```diff
@@ -26,6 +27,7 @@ data class OrganizationDto(
     val avatar: String? = null,
     val description: String = "",
     val canCreateContests: Boolean = false,
+    val canBulkUpload: Boolean = false,
```
Comment (kgevorkyan) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/OrganizationDto.kt:30]: `Bulk`?  what is the `Bulk`

may be `Vuln`?

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/Organization.kt [hunk 3]
```diff
@@ -29,6 +30,7 @@ data class Organization(
     var avatar: String? = null,
     var description: String? = null,
     var canCreateContests: Boolean = false,
+    var canBulkUpload: Boolean = false,
```
Comment (kgevorkyan) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/Organization.kt:33]: no, seems that it is the `Bulk`
please document this arg, thus

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/OrganizationDto.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/OrganizationDto.kt:30]: it is an array

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt]: Fixed it

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            li {
+                className = ClassName("list-group-item p-0 d-flex bg-light")
+                dragAndDropForm {
+                    isMultipleFilesSupported = true
+                    tooltipMessage = "Regular files/Executable files/ZIP Archives"
+                    onChangeEventHandler = { files ->
+                        setFilesForUploading(files!!.asList())
+                        uploadCosvFiles()
+                    }
+                }
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt:57]: only regular files (actually only JSON files)

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+val cosvFileManagerComponent: FC<Props> = FC { props ->
+    useTooltip()
+
+    val (cosvAvailableFiles, setCosvAvailableFiles) = useState<List<String>>(emptyList())
+
+    val (filesForUploading, setFilesForUploading) = useState<List<File>>(emptyList())
+
+    val uploadCosvFiles = useDeferredRequest {
+        val uploadedFiles: List<String> = post(
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt:29]: it's not uploaded files, it's uploaded ids

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt (hunk 5 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        ).decodeFromJsonString()
+        setCosvAvailableFiles { uploadedFiles }
+    }
+
+    div {
+        ul {
+            className = ClassName("list-group shadow")
+            cosvAvailableFiles.map { file ->
+                li {
+                    +file
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt:59]: Added

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt [hunk 7]
```diff
@@ -83,6 +83,16 @@ val organizationVulnerabilitiesTab: FC<OrganizationVulnerabilitiesMenuProps> = F
     }
     div {
         className = ClassName("col-8 mx-auto mt-1 mb-3")
+
+        div {
+            className = ClassName("d-flex justify-content-center mb-3")
+
+            withNavigate { navigateContext ->
+                buttonBuilder("Upload a new COSV files", style = "primary", isOutline = true) {
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt:91]: `Upload COSV files`

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    div {
+        ul {
+            className = ClassName("list-group shadow")
+            cosvAvailableFiles.map { file ->
+                li {
+                    className = ClassName("list-group-item")
+                    a {
+                        button {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt:50]: Why not `buttonBuilder`?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt (hunk 7 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt:91]: Suggestion to add translation here

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt [hunk 9]
```diff
@@ -138,6 +138,11 @@ val createVulnerabilityView: VFC = VFC {
                             className = ClassName("h4 mb-4")
                             +"Propose a new vulnerability"
                         }
+                        withNavigate { navigateContext ->
+                            buttonBuilder("Or upload a new COSV files", style = "primary", isOutline = true) {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt:142]: Translation

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+                div {
+                    className = ClassName("container card o-hidden border-0 shadow-lg my-2 card-body p-0 text-gray-900")
+                    style = jso {
+                        zIndex = (MAX_Z_INDEX - 1).unsafeCast<ZIndex>()
+                    }
+                    div {
+                        className = ClassName("p-5 text-center")
+                        h1 {
+                            className = ClassName("h4 mb-4")
+                            +"Upload a new COSV files"
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt:41]: Translation

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    val (cosvAvailableFiles, setCosvAvailableFiles) = useState<List<String>>(emptyList())
+
+    val (filesForUploading, setFilesForUploading) = useState<List<File>>(emptyList())
+
+    val uploadCosvFiles = useDeferredRequest {
+        val uploadedIds: List<String> = post(
+            url = "$apiUrl/cosv/batch-upload",
+            Headers(),
+            FormData().apply { filesForUploading.forEach { append(FILE_PART_NAME, it) } },
+            loadingHandler = ::noopLoadingHandler,
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt:38]: loading handler

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt (hunk 7 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt:91]: yes, please add translation also

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt (hunk 11 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt:38]: Either loading handler, or progress bar

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt (hunk 7 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt (hunk 8 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt (hunk 9 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/CreateVulnerabilityView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt (hunk 10 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/UploadVulnerabilityView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt (hunk 7 above)
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesTab.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt (hunk 11 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/fileuploader/CosvFileManagerComponent.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/i18next/locales/Translations.kt [hunk 12]
```diff
@@ -8,7 +8,8 @@ const val VULN_COLLECTION_RU = """
...
   "addNewMd": "|### Новые уязвимости\n    |Вы можете предложить свою [новую уязвимость](/vuln/create-vulnerability), если ее нет в нашем списке.\n    |После подтверждения со стороны администрации, она появится в архиве под уникальным ID.",
   "howToMd": "|### Как добавить уязвимость в проект?\n    |1. Создайте saveourtool [организацию](/create-organization) и [проект](/create-project);\n    |2. Откройте вкладку security вашего проекта;\n    |3. Создайте новую проблему и добавьте ее уникальный ID;\n    |\n    |В случае любых ошибок не стесняйтесь [связываться с нами](https://github.com/saveourtool/save-cloud/issues/new).",
-  "topRatingMd": "|### Рейтинговая система\n    |За каждую принятую уязвимость вы получаете очки рейтинга. Здесь вы можете ознакомиться с [рейтингом](/vuln/top-rating) пользователей и организаций."
+  "topRatingMd": "|### Рейтинговая система\n    |За каждую принятую уязвимость вы получаете очки рейтинга. Здесь вы можете ознакомиться с [рейтингом](/vuln/top-rating) пользователей и организаций.",
+  "Upload COSV files": "Загрузить COSV файлы"
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/i18next/locales/Translations.kt:12]: Why is it in `vulnerability-collection`? It is not VulnerabilityCollectionView

Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/i18next/locales/Translations.kt:12]: I did not want to create new sections for the sake of one line

**Pull request merged**